estela-web/src/pages/SpiderListPage/index.tsx

```diff
@@ -61,102 +77,150 @@ export class SpiderListPage extends Component<RouteComponentProps<RouteParams>,
-            key: "name",
-            render: (name: string, spider: SpiderList): ReactElement => (
-                <Link
-                    key={spider.key}
-                    to={`/projects/${this.projectId}/spiders/${spider.sid}`}
-                    className="text-estela-blue-medium"
-                >
-                    {name}
-                </Link>
-            ),
-        },
-        {
-            title: "LAST RUN",
-            dataIndex: "lastRun",
-            key: "lastRun",
-            render: (): ReactElement => <Text className="text-estela-black-medium text-xs">Not available</Text>,
-        },
-        {
-            title: "JOBS",
-            dataIndex: "jobs",
-            key: "jobs",
-            render: (): ReactElement => <Text className="text-estela-black-medium text-xs">-/-</Text>,
-        },
-    ];
+    getSpiderStatsAndUpdateDates = async (
+        startDate?: string | undefined | null,
+        endDate?: string | undefined | null,
+        sid?: number | undefined,
+    ): Promise<void> => {
+        this.setState({ loadedStats: false });
+        const { statsStartDate, statsEndDate } = this.state;
+        if (startDate && endDate) {
+            this.setState({
+                statsStartDate: moment.utc(startDate),
+                statsEndDate: moment.utc(endDate),
+            });
+        }
+        await this.apiService
+            .apiStatsSpiderList({
+                pid: this.projectId,
+                sid: String(sid),
+                startDate: !startDate ? statsStartDate.format("YYYY-MM-DD") : startDate,
+                endDate: !endDate ? statsEndDate.format("YYYY-MM-DD") : endDate,
+                offset: new Date().getTimezoneOffset(),
+            })
+            .then(
+                (response: SpidersStats[]) => {
+                    this.setState({
+                        spiderStats: [...response],
+                        loadedStats: true,
+                    });
+                },
+                (error: Error) => {
+                    notification.error({
+                        message: "No data",
+                        description: error.message,
+                    });
+                    this.setState({ loadedStats: true });
+                },
+            );
+    };
 
-    onPageChange = async (page: number): Promise<void> => {
-        this.setState({ loaded: false });
-        await this.getProjectSpiders(page);
+    onChangeDateRangeHandler: RangePickerProps["onChange"] = (_, dateStrings) => {
+        this.setState({ loadedStats: false });
+        const [startDateUTC, endDateUTC] = [
+            moment(dateStrings[0]).startOf("day").format("YYYY-MM-DD"),
+            moment(dateStrings[1]).endOf("day").format("YYYY-MM-DD"),
+        ];
+        this.getSpiderStatsAndUpdateDates(startDateUTC, endDateUTC, this.state.spiderId);
     };
 
+    onRefreshEventHandler = async () => {
+        await this.getSpiderStatsAndUpdateDates();
```

Comment:
When I do a refresh the page shows me `No data` notification, maybe the problem is here, because you're not adding the `sid` parameter to the function.
I left a video Clip showing the bug: https://bitmakerla.slack.com/files/U049CFELY5P/F05LQ11NW3W/screenshare_-_2023-08-07_8_21_46_am.webm